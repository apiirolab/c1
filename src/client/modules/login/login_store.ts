import { types } from 'mobx-state-tree';
import { FormStore } from '../form/models/form_store';

export const LoginStore = FormStore.named('LoginStore')
  .props({
    view: types.optional(types.enumeration('Views', ['login', 'forgot']), 'login'),
    user: '',
    password: ''
  })
  .actions(self => ({
    setView(view: 'login' | 'forgot') {
      self.view = view;
    }
  }));
