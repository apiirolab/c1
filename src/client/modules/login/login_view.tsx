import * as React from 'react';
import { Button, Form, Segment } from 'semantic-ui-react';

import styled from 'styled-components';
import { requiredValidator } from '../form/models/validation';
import { FormInput, InputView } from '../form/views/input_view';
import { LoginStore } from './login_store';

interface Props {
  store: typeof LoginStore.Type;
}

const CentredButton = styled(Button)`
  margin: 'auto';
`;

export const LoginView = ({ store }: Props) => (
  <Form>
    <Segment>
      <FormInput
        owner={store}
        label="User"
        name="user"
        icon="user"
        placeholder="Your Staff ID"
        fluid
        autoComplete="username"
        validators={[requiredValidator]}
      />
      <FormInput
        owner={store}
        label="Password"
        name="password"
        icon="lock"
        fluid
        autoComplete="current-password"
        type="password"
        validators={[requiredValidator]}
        placeholder="Your Password"
      />

      <div style={{ textAlign: 'center' }}>
        <CentredButton content="Login" primary icon="thumbs up" />
        <span style={{ position: 'absolute', right: '20px', marginTop: '6px' }}>
          Forgot Password?
        </span>
      </div>
    </Segment>
  </Form>
);
