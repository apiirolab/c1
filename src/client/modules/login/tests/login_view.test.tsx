import * as React from 'react';

import 'jest-styled-components';

import { create } from 'react-test-renderer';
import { LoginStore } from '../login_store';
import { LoginView } from '../login_view';

describe('Login', () => {
  describe('Default', () => {
    const store = LoginStore.create();

    const Login = <LoginView store={store} />;

    it('renders', () => {
      const component = create(Login);
      expect(component).toMatchSnapshot();
    });

    it('validates', () => {
      const component = create(Login);
      component.root
        .findAllByProps({ name: 'user' })[1]
        .props.onChange({ currentTarget: { value: '' } });

      component.root
        .findAllByProps({ name: 'password' })[1]
        .props.onChange({ currentTarget: { value: '' } });

      expect(component).toMatchSnapshot();

      component.root
        .findAllByProps({ name: 'user' })[1]
        .props.onChange({ currentTarget: { value: 'tomas' } });

      component.root
        .findAllByProps({ name: 'password' })[1]
        .props.onChange({ currentTarget: { value: 'tomas' } });

      expect(component).toMatchSnapshot();
    });

    return {
      component: Login
    };
  });
});
