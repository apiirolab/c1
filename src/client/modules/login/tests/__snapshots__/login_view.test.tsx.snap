// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login default renders 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <label>
        User
      </label>
      <div
        className="ui fluid icon input"
      >
        <input
          autoComplete="username"
          name="user"
          onChange={[Function]}
          placeholder="Your Staff ID"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label>
        Password
      </label>
      <div
        className="ui fluid icon input"
      >
        <input
          autoComplete="current-password"
          name="password"
          onChange={[Function]}
          placeholder="Your Password"
          type="password"
          value=""
        />
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="ui primary button sc-bdVaJa cLHDqx"
        onClick={[Function]}
        role="button"
      >
        <i
          aria-hidden="true"
          className="thumbs up icon"
        />
        Login
      </button>
      <span
        style={
          Object {
            "marginTop": "6px",
            "position": "absolute",
            "right": "20px",
          }
        }
      >
        Forgot Password?
      </span>
    </div>
  </div>
</form>
`;

exports[`Login default validates 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <label>
        User
      </label>
      <div
        className="ui error fluid icon input"
      >
        <input
          autoComplete="username"
          name="user"
          onChange={[Function]}
          placeholder="Your Staff ID"
          type="text"
          value=""
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
      <div
        className="ui red pointing label"
        onClick={[Function]}
      >
        This field is required!
      </div>
    </div>
    <div
      className="field"
    >
      <label>
        Password
      </label>
      <div
        className="ui error fluid icon input"
      >
        <input
          autoComplete="current-password"
          name="password"
          onChange={[Function]}
          placeholder="Your Password"
          type="password"
          value=""
        />
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </div>
      <div
        className="ui red pointing label"
        onClick={[Function]}
      >
        This field is required!
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="ui primary button sc-bdVaJa cLHDqx"
        onClick={[Function]}
        role="button"
      >
        <i
          aria-hidden="true"
          className="thumbs up icon"
        />
        Login
      </button>
      <span
        style={
          Object {
            "marginTop": "6px",
            "position": "absolute",
            "right": "20px",
          }
        }
      >
        Forgot Password?
      </span>
    </div>
  </div>
</form>
`;

exports[`Login default validates 2`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="ui segment"
  >
    <div
      className="field"
    >
      <label>
        User
      </label>
      <div
        className="ui fluid icon input"
      >
        <input
          autoComplete="username"
          name="user"
          onChange={[Function]}
          placeholder="Your Staff ID"
          type="text"
          value="tomas"
        />
        <i
          aria-hidden="true"
          className="user icon"
        />
      </div>
    </div>
    <div
      className="field"
    >
      <label>
        Password
      </label>
      <div
        className="ui fluid icon input"
      >
        <input
          autoComplete="current-password"
          name="password"
          onChange={[Function]}
          placeholder="Your Password"
          type="password"
          value="tomas"
        />
        <i
          aria-hidden="true"
          className="lock icon"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="ui primary button sc-bdVaJa cLHDqx"
        onClick={[Function]}
        role="button"
      >
        <i
          aria-hidden="true"
          className="thumbs up icon"
        />
        Login
      </button>
      <span
        style={
          Object {
            "marginTop": "6px",
            "position": "absolute",
            "right": "20px",
          }
        }
      >
        Forgot Password?
      </span>
    </div>
  </div>
</form>
`;
